src/python_testing/TC_MCORE_FS_1_5.py

```diff
@@ -304,7 +304,11 @@ async def test_TC_MCORE_FS_1_5(self):
                 endpoint, attribute, value = item['endpoint'], item['attribute'], item['value']
 
                 # Record arrival of an expected subscription change when seen
-                if endpoint == newly_added_endpoint and attribute == Clusters.AdministratorCommissioning.Attributes.WindowStatus:
+                if endpoint == newly_added_endpoint and attribute == cadmin_attr.WindowStatus:
+                    if value != th_server_direct_cadmin[cadmin_attr.WindowStatus]:
+                        logging.info("Window status is %r, waiting for %r", value,
+                                     th_server_direct_cadmin[cadmin_attr.WindowStatus])
+                        continue
```

Comment:
You might still need `time_remaining = report_waiting_timeout_delay_sec - elapsed` before continue

Reply:
Moved the elapsed/remaining bits into a `finally` block.